web-js/add-csv-file-as-featurelayer/index.html

```diff
@@ -0,0 +1,277 @@
+		const latitudeField = "latitude";
+		const longitudeField = "longitude";
+		const dateFields = ["time","updated"];
+		const startTimeField = "time";
+		const endTimeField = "";
+		const timeReference = {
+			timeZone: "UTC",
+			respectDaylightSaving: false
+		}
+
+        map = new Map("map", {
+          basemap: "gray",
+          center: [ -40, 0 ],
+          zoom: 2
+        });
+		
+		function getCSVFile(url) {
+			 esriRequest({
+				url: url,
+				content: {},
+				handleAs: "text"
+			 },{
+				usePost: false
+			 }).then((results) => {
+				parseFile(results);
+			 });
+		}
+		
+		function parseFile(results) {
+			let fields = [];
+			let features = [];
+			let fileLines = results.split(/\r?\n/);
+			let timeFieldIndexes = [];
+			let latitudeIndex = null;
+			let longitudeIndex = null;
+			let startTime = null;
+			let endTime = null;
+			
+			fields.push({
+				name: "OBJECTID",
+				type: "esriFieldTypeOID",
+				alias: "OID"						
+			});
+			
+			for(line in fileLines) {
+				// props to this guy: https://stackoverflow.com/questions/23582276/split-string-by-comma-but-ignore-commas-inside-quotes
+				let splitLine = fileLines[line].split(/,(?=(?:(?:[^"]*"){2})*[^"]*$)/);
+				splitLine = splitLine || [];
+				if(splitLine.length >= 3) {
+					if(line == 0) {
+						for(item in splitLine) {
+							fields.push({
+								name: splitLine[item],
+								alias: splitLine[item]
+							});
+							if(dateFields.indexOf(splitLine[item]) > -1) timeFieldIndexes.push(item);
+							else if(splitLine[item] == latitudeField) latitudeIndex = item;
+							else if(splitLine[item] == longitudeField) longitudeIndex = item;
+						}
+					}
+					else {
+						let latitude = null;
+						let longitude = null;
+						let attributes = {OBJECTID: features.length};
+						for(item in splitLine) {
+							let type = null;
+							if(timeFieldIndexes.indexOf(item) > -1) {
+								type = "esriFieldTypeDate";
+								splitLine[item] = Date.parse(splitLine[item]);
+								if(startTime && endTime) {
+									if(startTime > splitLine[item]) startTime = splitLine[item];
+									if(endTime < splitLine[item]) endTime = splitLine[item];
+								}
+								else {
+									startTime = splitLine[item];
+									endTime = splitLine[item];
+								}
+							}
+							else if(item == latitudeIndex) {
+								latitude = parseFloat(splitLine[item]);
+								type = "esriFieldTypeString";
+							}
+							else if(item == longitudeIndex) {
+								longitude = parseFloat(splitLine[item]);
+								type = "esriFieldTypeString";
+							}
+							else if(!isNaN(splitLine[item])) {
+								type = "esriFieldTypeDouble";
+								splitLine[item] = parseFloat(splitLine[item]);
+							}
+							else if(splitLine[item].toLowerCase() == "true") {
+								type = "esriFieldTypeBoolean";
+								splitLine[item] = true;
+							}
+							else if(splitLine[item].toLowerCase == "false") {
```

Comment:
@nhaney90 , looks great, just add the parentheses where we discussed while reviewing it.